Foundation/FileManager.swift

```diff
@@ -1385,25 +1388,27 @@ extension FileManager {
                     
                     // Skip the root.
                     _current = fts_read(stream)
-                    
                 }
 
                 _current = fts_read(stream)
                 while let current = _current {
+                    let filename = NSString(bytes: current.pointee.fts_path, length: Int(strlen(current.pointee.fts_path)), encoding: String.Encoding.utf8.rawValue)!._swiftObject
+                    let skipHidden = (_options.contains(.skipsHiddenFiles) && filename[filename._startOfLastPathComponent] == ".")
+
                     switch Int32(current.pointee.fts_info) {
                         case FTS_D:
-                            if _options.contains(.skipsSubdirectoryDescendants) {
+                            if skipHidden || _options.contains(.skipsSubdirectoryDescendants) || (_options.contains(.skipsPackageDescendants) && filename.hasSuffix(".app")) {
```

Comment:
The definitions of "isHidden" and "isPackage" should be abstracted away into some private functions or something. Eventually this abstraction should be URLResourceValues-based, but internal functions should suffice for now.

Perhaps even better, you could have a single function taking the path or filename and the enumeration options that decides whether the file should be hidden. This avoids the variable naming nit above.


That all said, since packages aren't an actual concept on Linux, so I feel that the option should be a no-op.

The . prefix definition of "hidden" seems fine, though incomplete on Darwin.